lib/hooks/useWeb3Wallet.ts

```diff
@@ -0,0 +1,173 @@
+import { WALLET_ADAPTERS, IProvider } from "@web3auth/base";
+import { Keyring } from "@polkadot/api";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { useWallet } from "lib/state/wallet";
+import { web3authAtom } from "lib/state/util/web3auth-config";
+import { useAtom } from "jotai";
+import { openloginAdapter, clientId } from "lib/state/util/web3auth-config";
+import { useNotifications } from "lib/state/notifications";
+interface loginOptions {
+  loginProvider: string;
+  extraLoginOptions: {
+    domain: string;
+    verifierIdField: "email" | "sub";
+    isVerifierIdCaseSensitive?: boolean;
+    login_hint?: string;
+    connection: string;
+  };
+}
+
+const auth0Domain = process.env.NEXT_PUBLIC_AUTH0_DOMAIN_ZTG;
+
+const useWeb3Wallet = () => {
+  const [web3auth] = useAtom(web3authAtom);
+  const notificationStore = useNotifications();
+  const { selectWallet, disconnectWallet, walletId } = useWallet();
+
+  const initWeb3Auth = async () => {
+    if (!clientId) {
+      return;
+    }
+    try {
+      if (web3auth) {
+        web3auth.configureAdapter(openloginAdapter);
+        await web3auth.init();
+        if (walletId === "web3auth" && web3auth.provider) {
+          await getKeypair(web3auth.provider);
+        }
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const login = async (loginOptions: loginOptions) => {
+    if (!web3auth || !auth0Domain) {
+      notificationStore.pushNotification(
+        `Error connecting: please try another login method or check back later.`,
+        {
+          type: "Error",
+          autoRemove: true,
+          lifetime: 5,
+        },
+      );
+      return;
+    }
+    if (web3auth.connected) {
+      await web3auth.logout();
```

Comment:
Might be a bit unexpected that a function that is called `login` also logs people out

Reply:
this is just in case for when a user is already logged in but wants to switch to another provider without logging out. if this isn't there it'll throw an error stating the provider is already connected